src/test/test_failure_slice.cpp

```
@@ -250,12 +253,11 @@ TEST_CASE("assume guard registers become relevant in slice", "[failure_slice][in
     ebpf_verifier_options_t options{};
     options.verbosity_opts.collect_instruction_deps = true;
     ElfObject elf{sample, options, &g_ebpf_platform_linux};
-    const auto& raw_progs = elf.get_programs("xdp");
-    REQUIRE(raw_progs.size() == 1);
-    auto prog_or_error = unmarshal(raw_progs.back(), options);
+    RawProgram raw_prog = elf.get_programs("xdp").back();
+    auto prog_or_error = unmarshal(raw_prog, options);
```

Cppcheck (2.20.0)

[style] The function 'cast_to_sint' is never used.

(unusedFunction)